tests/ui/rfc-2497-if-let-chains/chains-without-let.rs

```diff
@@ -1,7 +1,7 @@
 fn and_chain() {
     let z;
     if true && { z = 3; true} && z == 3 {}
-    //~^ ERROR E0381
+    // no longer ERROR E0381
 }
 
 fn and_chain_2() {
```

Comment:
Why is the test in this function unchanged?

Reply:
Sorry.  I don't understand the question.  It *is* changed. I removed the `~^` (meaning, it's no an error) and the output in `tests/ui/rfc-2497-if-let-chains/chains-without-let.stderr` shows that.

Reply:
Or do you mean I need to merge master to resolve the conflict?

Reply:
I mean in the other function `and_chain_2`.

Reply:
Because so far I'm only doing the `&&`s inside the test of `if`, not `&&` as expression.

I agree that that too should be fixed, along with the `||` cases.

If this branch gets merged, I will fix the other cases.